### PR TITLE
fix(#3002): memory management

### PR DESCRIPTION
**Pull request by maxonfjvipon**
What's done:
1. object `ram` is removed
2. object `heap` is renamed to `malloc` and improved
3. object `memory` uses `malloc`
4. fixed all the tests related to `malloc` and `memory`

Closes: #3002

---

## PR-Codex overview
The focus of this PR is to refactor and optimize memory handling in the codebase.

### Detailed summary
- Refactored memory allocation to `malloc` and `memory-block-pointer`
- Updated memory read and write operations
- Improved memory block freeing functionality

> The following files were skipped due to too many changes: `eo-runtime/src/main/java/EOorg/EOeolang/EOram$EOslice.java`, `eo-runtime/src/test/eo/org/eolang/while-tests.eo`, `eo-runtime/src/main/java/EOorg/EOeolang/EOram$EOwrite.java`, `eo-runtime/src/main/java/EOorg/EOeolang/EOram$EOram_slice$EOwrite.java`, `eo-runtime/src/test/eo/org/eolang/memory-tests.eo`, `/eo-runtime/src/main/java/EOorg/EOeolang/EOmemory$EO\317\206.java"`, `eo-runtime/src/test/eo/org/eolang/malloc-tests.eo`, `eo-runtime/src/test/java/EOorg/EOeolang/HeapsTest.java`, `eo-runtime/src/main/java/EOorg/EOeolang/Heaps.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@yegor256 please check

**Comment by maxonfjvipon** [comment]
@yegor256 please check

File: eo-runtime/src/main/java/org/eolang/Heaps.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Dynamic memory.
+ *
+ * @since 0.19
+ */
+@Versionized
+public final class Heaps {
```
Comment (yegor256) [inline eo-runtime/src/main/java/org/eolang/Heaps.java:38]: @maxonfjvipon move it to `EOeolang`

File: eo-runtime/src/main/eo/org/eolang/malloc.eo [hunk 2]
```diff
@@ -26,18 +26,33 @@
...
+  [id size] > memory-block-pointer
+    $ > pointer
 
-    # Write bytes to memory from position, according to the ram.slice object.
-    [data] > write /bytes
+    # Read `bytes` from the block in memory by the pointer.
+    [] > read /bytes
+
+    # Write data to the block in memory by the pointer.
+    [data] > write /true
```
Comment (yegor256) [inline eo-runtime/src/main/eo/org/eolang/malloc.eo:55]: @maxonfjvipon let's use `offset` as the first argument

**Comment by yegor256** [comment]
@maxonfjvipon see comments above

**Comment by maxonfjvipon** [comment]
@yegor256 please check again

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**